js/app.js

```diff
@@ -1,123 +1,92 @@
-service = angular.module("shoppingcart.services", ["ngResource"]);
+(function () {
+'use strict';
+
+angular.module('BlockonomicsApp', ['ngResource', 'monospaced.qrcode'])
+.controller('CryptoOptionsController', CryptoOptionsController)
+.controller('CheckoutController', CheckoutController)
+.factory('Order', Order)
+.config(Config);
+
+Config.$inject = ['$compileProvider'];
+function Config($compileProvider) {
+  $compileProvider.aHrefSanitizationWhitelist(/^\s*(https?|ftp|mailto|data|chrome-extension|bitcoin|bitcoincash):/);
+}
 
-service.factory('Order', function($resource) {
-    //There are two styles of callback url in 
-    //woocommerce, we have to support both
-    //https://docs.woocommerce.com/document/wc_api-the-woocommerce-api-callback/
-    var param = getParameterByNameBlocko('wc-api');
-    if (param)
-        param = {
-            "wc-api": param
-        };
+CryptoOptionsController.$inject = ['$scope', '$httpParamSerializer'];
+function CryptoOptionsController($scope, $httpParamSerializer) {
+    var active_cryptos_div = document.getElementById("active_cryptos");
+    var active_cryptos = JSON.parse(active_cryptos_div.dataset.active_cryptos);
+    $scope.no_display_error = true;
+    $scope.active_cryptos = active_cryptos;
+    $scope.crypto_selecter  = true;
+    //fetch url params
+    $scope.order_id = getParameterByNameBlocko("select_crypto");
+
+    $scope.checkout_order_url = function() {
```

Comment:
There is visible code duplication in *checkout_order_url* and *finish_order_url* methods. Re factor them to a single method. *get_wc_endpoint* .. it takes dict parameters are input and returns a wc URL. No need to use $scope variable inside these methods.

Reply:
Refactored in 1f5a634